### PR TITLE
implement List db method for webhooks

### PR DESCRIPTION
**Pull request by mollylogue**
Closes https://github.com/sourcegraph/sourcegraph/issues/42740

## Test plan
New unit tests

### CODE REVIEW

File: internal/database/webhooks.go [hunk 1]
```diff
@@ -246,11 +246,42 @@ RETURNING
...
+	return res, nil
 }
 
+type WebhookListOptions struct {
+	Kind string
+	*LimitOffset
+}
+
+const webhookListQueryFmtstr = `
+-- source: internal/database/webhooks.go:List
```
Comment (sashaostrikov) [inline internal/database/webhooks.go:279]:
```suggestion
```
These comments recently became obsolete, pls see https://github.com/sourcegraph/sourcegraph/pull/42876

File: internal/database/webhooks.go [hunk 2]
```diff
@@ -246,11 +246,42 @@ RETURNING
...
+	}
+	if opt.LimitOffset != nil {
+		q = sqlf.Sprintf("%s\n%s", q, opt.LimitOffset.SQL())
+	}
+	rows, err := s.Query(ctx, q)
+	if err != nil {
+		return []*types.Webhook{}, errors.Wrap(err, "error running query")
+	}
+	defer rows.Close()
+	res := make([]*types.Webhook, 0, 10)
```
Comment (sashaostrikov) [inline internal/database/webhooks.go:262]: I'm curious, why 10? :)

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: internal/database/webhooks.go (hunk 2 above)
Comment (mollylogue) [reply internal/database/webhooks.go:262]: Ah, I meant to actually delete/update this. I was using another db method as a template and that was the default. I think we were planning to have the default pagination be 20, so it'd probably make the most sense to set this to 20.

File: internal/database/webhooks.go (hunk 1 above)
Comment (mollylogue) [reply internal/database/webhooks.go]: 👍

**Pull request merged**